File: api/src/main/java/com/agenda/agendacontatos/model/Contato.java
package com.agenda.agendacontatos.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "contato")
public class Contato {
    
    @Id
    @Column ( name = "email")
    private String email;

    @Column ( name = "nome")
    private String nome;

    @Column ( name = "senha")
    private String senha;

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getSenha() {
        return senha;
    }

    public void setSenha(String senha) {
        this.senha = senha;
    }

}

File: lib/Contato.dart
class Contato {
  final String nome;
  final String email;
  final String senha;

  Contato({
    required this.nome,
    required this.email,
    required this.senha,
  });

  Map<String, dynamic> toJson() {
    return {
      'nome': nome,
      'email': email,
      'senha': senha,
    };
  }
}
